probe/endpoint/resolver_test.go

```diff
@@ -0,0 +1,40 @@
+package endpoint_test
+
+import (
+	"testing"
+
+	"github.com/weaveworks/scope/probe/endpoint"
+	"time"
+)
+
+func TestReverseResolver(t *testing.T) {
+	revRes := endpoint.NewReverseResolver(100)
+	tests := []struct {
+		ip   string
+		name string
+	}{
+		{"8.8.8.8", "google-public-dns-a.google.com."},
+		{"8.8.4.4", "google-public-dns-b.google.com."},
+	}
+
+	// first time: no names are returned for our reverse resolutions
+	for _, v := range tests {
+		if name, err := revRes.Get(v.ip); name != "" || err == nil {
+			t.Errorf("we didn't get an error, or the cache was not empty, hen trying to resolve '%s'", v.ip)
+		}
+	}
+
+	// after some time, the async resolver should have the names in the cache
+	time.Sleep(500 * time.Millisecond)
```

Comment:
Hmm, can we make this deterministic, to avoid unit tests sleeping for half a second? That may require extra hooks in the reverse resolver...